src/dialect/mod.rs

```diff
@@ -157,6 +157,14 @@ pub trait Dialect: Debug + Any {
         // return None to fall back to the default behavior
         None
     }
+    /// Does the dialect support trailing commas around the query?
+    fn supports_trailing_commas(&self) -> bool {
```

Comment:
`supports_trailing_commas` suggests to me that it is supporting trailing commas everywhere, and `supports_projection_trailing_commas` suggests just for projection

If `supports_trailing_commas`  means just for queries, can you please name it to something more specific?

Reply:
This option should try to allow trailing commas everywhere. I'm not really following what you mean by "just for queries"

Reply:
I noticed something while going over the tests, Create table by default allows trailing commas, because it doesn't use the `parse_comma_separated` function, which was added later.